templates/model_package/set_resource_name_as_arn.go.tpl

```diff
@@ -0,0 +1,9 @@
+    // If ModelPackageName not set after newRequestPayload, attempt to use ARN
+    // This is because versioned modelpackage uses ARN not name
+	if input.ModelPackageName == nil {
+		arn := r.Identifiers().ARN()
+		if arn == nil {
+			return nil, ackerr.NotFound
+		}
+		input.SetModelPackageName(string(*arn))
```

Comment:
ModelPackageName is internal input struct name for the field that takes in either an ARN or the ModelPackageName. Naming of the field is poor since ModelPackageName is an existing field which causes confusion